### PR TITLE
Calendly integration for dashboard publish

### LINKED ISSUE #1733: implement "share" button on dashboard to point to "contact us" flow
see https://www.notion.so/rilldata/Publish-Button-on-Developer-Dashboard-to-closed-beta-form-1ff589e13cb34388b621b32799f6ad6b

### PR DESCRIPTION
**Pull request by AdityaHegde**
closes #1733

### CODE REVIEW

File: web-common/src/features/dashboards/workspace/DashboardHeader.svelte [hunk 1]
```diff
@@ -69,8 +83,14 @@
             Edit this dashboard's metrics & settings
           </TooltipContent>
         </Tooltip>
-      </div>
-    {/if}
+      {/if}
+      <Tooltip distance={8}>
+        <Button on:click={openCalenderly} type="primary">Publish</Button>
```
Comment (ericpgreen2) [inline web-common/src/features/dashboards/workspace/DashboardHeader.svelte:88]: This should be behind the `isEditableDashboard` flag too. The read-only dashboards will be hosted/published already.

File: web-common/src/features/dashboards/workspace/DashboardHeader.svelte [hunk 2]
```diff
@@ -38,6 +40,18 @@
   $: metaQuery = useMetaQuery($runtimeStore.instanceId, metricViewName);
   $: displayName = $metaQuery.data?.label;
   $: isEditableDashboard = $runtimeStore.readOnly === false;
+
+  function openCalenderly() {
```
Comment (ericpgreen2) [inline web-common/src/features/dashboards/workspace/DashboardHeader.svelte:44]: typo: "openCalendly" not "openCalenderly"

File: web-common/src/features/dashboards/workspace/DashboardHeader.svelte [hunk 3]
```diff
@@ -49,18 +63,18 @@
...
     <!-- title element -->
     <h1 style:line-height="1.1" style:margin-top="-1px">
       <div class="pl-4" style:font-family="InterDisplay" style:font-size="20px">
         {displayName || metricViewName}
       </div>
     </h1>
-    <!-- top right CTAs -->
-    {#if isEditableDashboard}
-      <div style="flex-shrink: 0;">
+    <PanelCTA>
```
Comment (ericpgreen2) [inline web-common/src/features/dashboards/workspace/DashboardHeader.svelte:75]: This triggers a `svelte-check` error: "Property 'side' is missing in type '{}' but required in type '{ side: \"left\" | \"right\"; }'."

**Comment by ericpgreen2** [comment]
Regarding the telemetry changes, I'll let @magorlick do the thorough review, but one idea is to add a schema version to each event type.

**Comment by AdityaHegde** [comment]
Thanks @ericpgreen2 . Good point about the schema. Feel like it fits the other PR https://github.com/rilldata/rill-developer/pull/1760 more than this.

**Comment by ericpgreen2** [comment]
Oh I hadn't seen 1760 – agreed.

I'll approve from the code standpoint, but I imagine @magorlick wants to review UX.

**Review by ericpgreen2 — APPROVED** [review]

File: web-common/src/components/modal/dialog/Dialog.svelte [hunk 4]
```diff
@@ -21,37 +21,48 @@
...
+        yDimClasses = "h-1/2";
+        break;
+
+      case "lg":
+        xDimClasses = "w-4/5 md:w-3/5 xl:w-1/2 2xl:w-1/3";
+        yDimClasses = "h-3/5";
+        break;
+
+      case "full":
+        width = "calc(100% - 2*var(--header-height))";
```
Comment (magorlick) [inline web-common/src/components/modal/dialog/Dialog.svelte:46]:
```suggestion
        xDimClasses = "w-4/5 md:w-3/5 xl:w-1/2 2xl:w-2/3";
```

File: web-common/src/components/modal/dialog/Dialog.svelte [hunk 5]
```diff
@@ -21,37 +21,48 @@
...
+        height = "calc(100% - 2*var(--header-height))";
```
Comment (magorlick) [inline web-common/src/components/modal/dialog/Dialog.svelte:47]:
```suggestion
        yDimClasses = "h-4/5";
```

**Review by magorlick — CHANGES_REQUESTED** [review]
Thank you for these changes @AdityaHegde ! The functionality is great, we some refinement needed for the experience and a telemetry check.

**[p0]** Please update copy for tool tip for publish button to read, ***“Schedule time to chat with Rill about early access to hosted dashboards.”***

**[p0]** Make the modal window tighter around the calendly embed. It should have no more than 60px of padding around the embed at all sizes. I left a suggestion that might work, but I am not sure what else we are using "full" for in Dialog.svelte.

**[p0]** I couldn’t see telemetry for the publish button successfully send. Not sure if that should be what is happening here. 

Internal notes here: https://www.notion.so/rilldata/1756-calendly-for-publish-a9747bc61feb4f3faf1ed9bb7084f52e

**Comment by AdityaHegde** [comment]
@magorlick Have made the changes you suggested. I am not an expert in css, so lemme know if it looks good.

The telemetry is firing for me. It fires when you open the calendly modal and after you have successfully scheduled a meeting. Lemme know if this works.

**Comment by magorlick** [comment]
@AdityaHegde , I can see the start and success telemetry now and the felt experience is better. Very close!

The calendly link is no longer correct. Please update to the one I am DMing you, also in our docs. :)

**Comment by AdityaHegde** [comment]
@magorlick Updated the link.

**Review by magorlick — APPROVED** [review]
lgtm!

**Pull request merged**